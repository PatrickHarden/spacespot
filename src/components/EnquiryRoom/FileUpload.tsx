import React, { useCallback, ReactNode } from 'react'
import { useDropzone } from 'react-dropzone'

export const onDropCB = (
  addFile: (file: File, reader: FileReader) => () => void,
) => (acceptedFiles: Array<File>) => {
  const file = acceptedFiles[0]
  if (!file) {
    return
  }
  const reader = new FileReader()
  reader.onloadend = addFile(file, reader)
  reader.readAsDataURL(file)
}

const FileUpload = (props: {
  className?: string
  children: ReactNode
  accept: string
  addFile: (file: File, reader: FileReader) => () => void
}) => {
  const { children, className, accept, addFile } = props
  const onDrop = useCallback(onDropCB(addFile), [addFile])
  const { getRootProps, getInputProps, isDragActive } = useDropzone({
    onDrop,
    accept,
    maxSize: 10000000,
  })
  return (
    <div className={className}>
      <div {...getRootProps()} className={isDragActive ? 'isDragging' : ''}>
        <input {...getInputProps()} />
        {children}
      </div>
    </div>
  )
}

export default FileUpload
