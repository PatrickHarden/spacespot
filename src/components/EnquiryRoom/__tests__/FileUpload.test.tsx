import React from 'react'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import FileUpload, { onDropCB } from '../FileUpload'

it('Renders docs', () => {
  const wrapper = mount(
    <FileUpload accept="application/pdf" addFile={jest.fn()}>
      Add file
    </FileUpload>,
  )
  expect(mountToJson(wrapper)).toMatchSnapshot()

  const addFile = () => jest.fn()
  const cb1 = onDropCB(addFile)
  const file = new Blob(['test'], { type: 'pdf' }) as File
  const acceptedFiles = [file]
  cb1([])
  cb1(acceptedFiles)
})
