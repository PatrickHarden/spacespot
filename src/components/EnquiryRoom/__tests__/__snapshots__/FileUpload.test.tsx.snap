// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders docs 1`] = `
<FileUpload
  accept="application/pdf"
  addFile={[MockFunction]}
>
  <div>
    <div
      className=""
      onBlur={[Function]}
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      <input
        accept="application/pdf"
        autoComplete="off"
        multiple={true}
        onChange={[Function]}
        onClick={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        tabIndex={-1}
        type="file"
      />
      Add file
    </div>
  </div>
</FileUpload>
`;
